driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -469,13 +469,13 @@ public void onSet(Connection connection, Message.Response response, long latency
                     break;
             }
         } catch (Exception e) {
+            exceptionToReport = e;
             setFinalException(connection, e);
         } finally {
             if (releaseConnection && connection instanceof PooledConnection)
                 ((PooledConnection)connection).release();
-
-            if (queriedHost != null)
-                manager.cluster.manager.reportLatency(queriedHost, latency);
+            if (queriedHost != null && statement != Statement.DEFAULT)
```

Comment:
I decided to exclude internal statements from being reported to clients because Statement.DEFAULT is not public, so clients would not be able to do anything useful with it anyway.